draft-ietf-jsonpath-base.md

```diff
@@ -1485,87 +1494,44 @@ Notes:
 * `LogicalType` is an abstraction of the result of a `logical-expr`.
   Its two instances, `LogicalTrue` and `LogicalFalse`, are not related to
   the JSON literals `true` and `false` and have no syntactical representation in JSONPath.
+* `SingleNodeType` is an abstraction of a `singular-path` (which appears in a comparison,
+  in a test expression, or as a function argument).
 * `NodesType` is an abstraction of a `filter-path` (which appears
   in a test expression or as a function argument).
   Members of `NodesType` have no syntactical representation in JSONPath.
+* `SingleNodeType` is a subtype of `NodesType`. An instance of `SingleNodeType` is well-typed
+  wherever an instance of `NodesType` is well-typed.
 
 The abstract instances above can be obtained from the concrete representations in {{tbl-typerep}}.
 
 | Abstract Instance | Concrete Representations                                           |
 | :---------------: | :----------------------:                                           |
 | `Value(v)`        | JSON value `v`                                                     |
 | `Nothing`         | A representation of the absence of a JSON value, distinct from the JSON literal `null`, e.g., from a Singular Path or `filter-path` resulting in an empty nodelist   |
-| `Nodes(nl)`       | A list of zero or more nodes, e.g., from a `filter-path` resulting in the nodelist `nl`, which may or may not be empty  |
+| `Nodes(nl)`       | A list of zero or more nodes, e.g., from a `filter-path` or a `singular-path` resulting in the nodelist `nl`, which may or may not be empty  |
```

Comment:
We may need to add `Nodes(nl1)` (or maybe just `nl1`) since that's used as the abstraction for `SingleNodeType`

Reply:
I tried adding another row along those lines but I found it very confusing. A `filter-path` can produce a nodelist with one node, so an abstract instance of the form `Nodes(nl1)` doesn't necessarily belong to the `SingleNodeType`. We use types to work out what values can occur, not vice versa.

I'm concerned when you say "... `Nodes(nl1)` ... [is] used as the abstraction for `SingleNodeType`